sky/serve/replica_managers.py

```diff
@@ -307,22 +352,39 @@ def to_replica_status(self) -> serve_state.ReplicaStatus:
         if self.service_ready_now:
             # Service is ready
             return serve_state.ReplicaStatus.READY
-        if self.first_ready_time is not None:
+        if self.first_ready_time is not None and self.first_ready_time != -1:
             # Service was ready before but not now
             return serve_state.ReplicaStatus.NOT_READY
         else:
             # No readiness probe passed and sky.launch finished
             return serve_state.ReplicaStatus.STARTING
 
+    def json(self) -> Dict[str, Any]:
+        return {
+            'sky_launch_status': self.sky_launch_status.value if
+                                 self.sky_launch_status is not None else None,
+            'user_app_failed': self.user_app_failed,
+            'service_ready_now': self.service_ready_now,
+            'first_ready_time': self.first_ready_time,
+            'sky_down_status': self.sky_down_status.value
+                               if self.sky_down_status is not None else None,
+            'preempted': self.preempted,
+        }
+
 
 class ReplicaInfo:
     """Replica info for each replica."""
 
-    def __init__(self, replica_id: int, cluster_name: str,
-                 replica_port: str) -> None:
+    _VERSION = 2
```

Comment:
```suggestion
    _VERSION = 1
```
we should start as version 1?